doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,418 @@ Compliant example::
+b) For functions whose return value is of a pointer type, the `*` character
+   shall be right before the function identifyer with no spaces in between in the
+   function prototype.
+
+Compliant example::
+
+    uint32_t *showcase_ptr;
+    uint32_t *showcase_func(void);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t* showcase_ptr;
+       uint32_t* showcase_func(void);
+
+
+WS-06: A single space shall exist around binary and ternary operators
+---------------------------------------------------------------------
+
+This rule applies to all binary arithmetic, bit-wise, logical, relational,
+equality and assignment operators, as well as the ternary conditional operator.
+
+Compliant example::
+
+    uint32_t showcase = 32U;
+    
+    showcase = showcase * 2U;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase=32U;
+       
+       showcase=showcase*2U;
+
+
+WS-07: Space shall not exist after unary operator
+-------------------------------------------------
+
+There shall be no space between a unary operator and its operand. This rule
+applies to array subscripting, function calls, member accesses, prefix or
+postfix increments and decrements, address and indirection operators.
+
+Compliant example::
+
+    int *x;
+    int y = y + *x;
+    int a = b->member;
+    int c = d[3];
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       int * x;
+       int y = y + * x;
+       int a = b ->member;
+       int c = d [3];
+
+
+WS-08: A single space shall exist right after semicolons in for-loop headers
+----------------------------------------------------------------------------
+
+A single space shall exist right after semicolons that separete the different
+expressions in for-loop headers.
+
+Compliant example::
+
+    uint32_t i;
+    
+    for (i = 0U; i < 5U; i++) {
+        printf("count: %d \n", i);
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t i;
+       
+       for (i = 0U;i < 5U;i++) {
+           printf("count: %d \n", i);
+       }
+
+
+
+Braces
+======
+
+BC-01: Braces after if/switch/for/do/while shall be in the same line
+--------------------------------------------------------------------
+
+The statement after if/switch/for/do/while shall always be a compound statement
+with its opening brace in the same line as the keyword.
+
+Compliant example::
+
+    uint32_t numerator = 32U;
+    uint32_t denominator = 0U;
+    uint32_t quotient;
+    
+    if (denominator != 0U) {
+        quotient = numerator / denominator;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t numerator = 32U;
+       uint32_t denominator = 0U;
+       uint32_t quotient;
+       
+       if (denominator != 0U)
+       {
+           quotient = numerator / denominator;
+       }
+
+
+BC-02: Newline and brace shall be after function prototypes
```

Comment:
Actually, the rule description reads better than this rule, so I'd use that instead.